backend/apiPackage.ts

```diff
@@ -67,4 +67,78 @@ export async function getPackagesByName(req : Request, res: Response){
   }catch(error){
     return res.status(500).send(`Error in getPackagesByName: ${error}`);
   }
+}
+
+// Packagedownloadresponsetype
+export type PackageDownloadResponseType = {
+  metadata: apiSchema.PackageMetadata;
+  data: apiSchema.PackageData;
+};
+
+// For: get package download
+export async function getPackageDownload(req: Request, res: Response) {
+  try {
+      if (req.query?.name === undefined) {
+          return res.status(400).send(`Error in getPackageDownload: Package name is undefined`);
+      }
+      if (req.query?.version === undefined) {
+          return res.status(400).send(`Error in getPackageDownload: Package version is undefined`);
+      }
+
+      const packageName = req.query.name as string;
+      const packageVersion = req.query.version as string;
+
+      const dbPackage = await prismaCalls.getPackageWithMetadataAndData(packageName, packageVersion);
+
+      if (!dbPackage || !dbPackage.data || !dbPackage.metadata) {
+          return res.status(404).send(`Error in getPackageDownload: Package not found`);
+      }
+
+      const apiPackageData: apiSchema.PackageData = {
+          Content: dbPackage.data.content,
+          URL: dbPackage.data.URL,
+          JSProgram: dbPackage.data.JSProgram,
+      };
+
+      const apiPackageMetadata: apiSchema.PackageMetadata = {
+          Name: dbPackage.metadata.name,
+          Version: dbPackage.metadata.version,
+          ID: dbPackage.metadata.id,
+      };
+
+      return res.status(200).json({ metadata: apiPackageMetadata, data: apiPackageData });
+  } catch (error) {
+      return res.status(500).send(`Error in getPackageDownload: ${error}`);
+  }
+}
+
+// Packagedownloadresponsetype
+export type PackageUpdateRequestType = {
+  name: string;
+  version: string;
+  content: string;
+  URL: string;
+  JSProgram: string;
+};
```

Comment:
use the package type from the apiSchema.ts file. And also, you're missing the ID from this anyway